### PR TITLE
REL-2992: GMOS Hamamatsu CCD ITC overestimates the peak signal + background 

### PR DESCRIPTION
**Pull request by cquiroz**
The OT ITC spectroscopy table was ignoring the fact that some instruments have multiple CCDs. This PR fixes that taking the maximum of the CCD values as requested  and improves the safety of the calculations

### CODE REVIEW

File: bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala [hunk 1]
```diff
@@ -35,50 +35,54 @@ final case class ItcCcd(
...
 sealed trait ItcResult extends Serializable {
-  def ccds:                         List[ItcCcd]
+  def ccds: List[ItcCcd]
 
-  def ccd(i: Int = 0)               = ccds(i % ccds.length)
-  def peakPixelFlux(ccdIx: Int = 0) = ccd(ccdIx).peakPixelFlux.toInt
+  /**
+    * Max value for a property across the CCDs
+    */
+  private def maxP[A](f: ItcCcd => A)(implicit a: scala.Ordering[A]): A = ccds.map(f).max
```
Comment (swalker2m) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: Could also be: `ccds.maxBy(f)`.

File: bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala [hunk 2]
```diff
@@ -35,50 +35,58 @@ final case class ItcCcd(
...
 
-
 // === SPECTROSCOPY RESULTS
 
 // There are two different types of charts
 sealed trait SpcChartType
-case object SignalChart       extends SpcChartType { val instance = this } // signal and background over wavelength [nm]
-case object S2NChart          extends SpcChartType { val instance = this } // single and final S2N over wavelength [nm]
-case object SignalPixelChart  extends SpcChartType { val instance = this } // single and final S2N over wavelength [nm]
+case object SignalChart       extends SpcChartType { val instance: SpcChartType = this } // signal and background over wavelength [nm]
```
Comment (swalker2m) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: The `instance` methods are done for the benefit of Java access, I'm guessing?

File: bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala (hunk 1 above)
Comment (cquiroz) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: doesn't `mabBy` return the maximum `ItcCcd` based on some property? I need the max value of the property `A`

Comment (swalker2m) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: Ah, right.  It'd have to be `f(ccds.maxBy(f))`.  Saves a pass though the `List`, I guess, but probably not worth it.

Comment (swalker2m) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: We know for sure that the list isn't empty, by the way?

Comment (cquiroz) [inline bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala]: This is a pain point on this. The list could be empty, but not in any meaningful use case. I tried using `NonEmptyList` but I hit a wall being non serializable and it lead to a bunch of broken tests. I'd like to make it use `NEL` but it is probably not worth the pain inflicted at this moment

**Pull request merged**